torchtune/training/checkpointing/_checkpointer.py

```diff
@@ -343,26 +362,15 @@ def __init__(
         adapter_checkpoint: Optional[str] = None,
         recipe_checkpoint: Optional[str] = None,
         resume_from_checkpoint: bool = False,
-        safe_serialization: bool = False,
+        safe_serialization: bool = True,
```

Comment:
are we ok with changing the default? I think it has some performance benefits when working with hf